airbyte-integrations/connectors/source-commercetools/source_commercetools/source.py

```diff
@@ -0,0 +1,199 @@
+        if offset+self.limit < total:
+            return dict(offset=offset+self.limit)
+        else:
+            return None
+
+    @property
+    def url_base(self) -> str:
+        return f"https://api.{self.region}.commercetools.com/{self.project_key}/"
+
+    def path(self, **kwargs) -> str:
+        return f"{self.data_field}"
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {"offset": self.offset, "limit": self.limit}
+        if next_page_token:
+            params.update(**next_page_token)
+        params.update({"sort":"lastModifiedAt asc"})
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        json_response = response.json()
+        records = json_response.get("results", []) if json_response is not None else []
+        yield from records
+
+
+# Basic incremental stream
+class IncrementalCommercetoolsStream(CommercetoolsStream, ABC):
+
+    @property
+    def limit(self):
+        return super().limit
+
+    # Setting the default cursor field for all streams
+    cursor_field = "lastModifiedAt"
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        return {self.cursor_field: max(latest_record.get(self.cursor_field, ""), current_stream_state.get(self.cursor_field, ""))}
+
+    def request_params(self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs):
+        params = super().request_params(stream_state=stream_state, next_page_token=next_page_token, **kwargs)
+        # If there is a next page token then we should only send pagination-related parameters.
+        if not next_page_token:
+            params["limit"] = self.limit
+            params["sort"] = "lastModifiedAt asc"
+            if stream_state:
+                params["where"] = "lastModifiedAt >= :date"
+                params["var.date"] = stream_state.get(self.cursor_field)
+        return params
+
+    # Parse the stream_slice with respect to stream_state for Incremental refresh
+    # cases where we slice the stream, the endpoints for those classes don't accept any other filtering,
+    # but they provide us with the updated_at field in most cases, so we used that as incremental filtering during the order slicing.
+    def filter_records_newer_than_state(self, stream_state: Mapping[str, Any] = None, records_slice: Mapping[str, Any] = None) -> Iterable:
+        # Getting records >= state
+       if stream_state:
+           for record in records_slice:
+               if record[self.cursor_field] >= stream_state.get(self.cursor_field):
+                   yield record
+       else:
+           yield from records_slice
+
+
+class Customers(IncrementalCommercetoolsStream):
+    data_field = "customers"
```

Comment:
Is possible to add the URL for this endpoint?

Reply:
@marcosmarxm this class is a subclass of class that has the path already set up by using the data_field property so there was no need to add it here

Reply:
Sorry, the doc URL